src/components/standup/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { FC, useState, useEffect } from 'react';
+import moment from 'moment';
+import styles from '@/components/standup/standupContainer.module.scss';
+
+import {
+    useSaveProgressMutation,
+    useUserProgressDetailsQuery,
+} from '@/app/services/progressesApi';
+import { useGetUserQuery } from '@/app/services/userApi';
+
+import FormInputComponent from './FormInputComponent';
+import { standupUpdateType } from '@/types/standup.type';
+import { getTotalMissedUpdates } from '@/utils/getTotalMissedUpdate';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import {
+    ERROR_MESSAGE,
+    STANDUP_SUBMISSION_SUCCESS,
+} from '@/constants/constants';
+import ProgressHeader from '../ProgressForm/ProgressHeader';
+
+const defaultState = {
+    type: 'user',
+    completed: '',
+    planned: '',
+    blockers: '',
+};
+
+const StandUpContainer: FC = () => {
+    const [standupUpdate, setStandupUpdate] =
+        useState<standupUpdateType>(defaultState);
+
+    const [buttonDisable, setButtonDisable] = useState<boolean>(true);
+    const [addStandup] = useSaveProgressMutation();
+    const { data: user } = useGetUserQuery();
+    const { data: userStandupdata } = useUserProgressDetailsQuery(user?.id);
+
+    const { SUCCESS, ERROR } = ToastTypes;
+    const standupDates = userStandupdata?.data?.map((element) => element.date);
+    const totalMissedUpdates = getTotalMissedUpdates(standupDates || []);
+    const yesterdayDate = moment().subtract(1, 'days').format('MMMM DD, YYYY');
+
+    const buttonStyleClass = buttonDisable
+        ? `${styles.nonActiveButton}`
+        : `${styles.activeButton}`;
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setStandupUpdate((prevStandupUpdate) => ({
+            ...prevStandupUpdate,
+            [event.target.name]: event.target.value,
+        }));
+    };
+    const isValidate = () => {
+        return (
+            standupUpdate.completed !== '' &&
+            standupUpdate.planned !== '' &&
+            standupUpdate.blockers !== ''
+        );
+    };
+    useEffect(() => {
+        const isValid = isValidate();
+        setButtonDisable(!isValid);
+    }, [standupUpdate]);
```

Comment:
Since standupUpdate is a state and on input that state will be updated, won't the component reload? Won't it have an effect on the standupComponent? if it will have an effect do we need this useEffect?

context: I can see on every update in formInput component the state of this component is being updated via handleChange function written in line 46

Reply:
fixed

Reply:
Could you let me know what did you fix here? I don't see any code changes on it.